csi/moac/volume.js

```diff
@@ -562,6 +570,12 @@ class Volume {
         `Shrinking the volume "${this}" is not supported`
       );
     }
+    if (this.protocol !== spec.protocol) {
+      throw new GrpcError(
+        GrpcCode.INVALID_ARGUMENT,
+        `Changing the protocol for volume "${this}" is not supported`
+      );
+    }
```

Comment:
Can you add a test case for this case that ends up with an exception?

Reply:
:+1: